src/NServiceBus.Core.Tests/Msmq/MsmqQueueCreatorTests.cs

```diff
@@ -247,12 +229,16 @@ void DeleteQueueIfPresent(string queueName)
         {
             var path = MsmqAddress.Parse(queueName).PathWithoutPrefix;
 
-            if (!MessageQueue.Exists(path))
+            if (MessageQueue.Exists(path))
             {
-                return;
+                MessageQueue.Delete(path);
             }
+        }
 
-            MessageQueue.Delete(path);
+        string GetReallyLongQueueName()
+        {
+            var maxQueueNameForSetPermissionToWork = 102 - Environment.MachineName.Length;
```

Comment:
The thinking was that since we always call `CreateQueueIfNecessary` when installers are run (and we tell users to run them on every deploy of the endpoint) we need to make sure that it won't choke for existing queues since it would prevent existing users with to long queues to upgrade their endpoints. 

Does that logic make sense?

If yes is the implementation correct based on the above?